Essentials/src/main/java/com/earth2me/essentials/MetaItemStack.java

```diff
@@ -216,11 +217,22 @@ public void addStringMeta(final CommandSource sender, final boolean allowUnsafe,
             } else {
                 throw new Exception(tl("onlyPlayerSkulls"));
             }
-        } else if (split.length > 1 && split[0].equalsIgnoreCase("book") && MaterialUtil.isEditableBook(stack.getType()) && (hasMetaPermission(sender, "book",true, true, ess) || hasMetaPermission(sender, "chapter-" + split[1].toLowerCase(Locale.ENGLISH), true, true, ess))) {
+        } else if (split.length > 1 && split[0].equalsIgnoreCase("book") && MaterialUtil.isEditableBook(stack.getType()) && (hasMetaPermission(sender, "book", true, true, ess) || hasMetaPermission(sender, "chapter-" + split[1].toLowerCase(Locale.ENGLISH), true, true, ess))) {
             final BookMeta meta = (BookMeta) stack.getItemMeta();
             final IText input = new BookInput("book", true, ess);
             final BookPager pager = new BookPager(input);
-
+            // This fix only applies to written books.
+            if (stack.getType() == WRITTEN_BOOK) {
+                // Written books require an author and a title. https://bugs.mojang.com/browse/MC-59153
+                if (!meta.hasAuthor()) {
+                    meta.setAuthor(sender.getDisplayName());
```

Comment:
Particular reason for this being display name rather than username?

Reply:
I would also prefer we used the username here.